ethcore/src/engines/clique/mod.rs

```diff
@@ -633,42 +672,40 @@ impl Engine<EthereumMachine> for Clique {
 	}
 
 	// Our task here is to set difficulty
+	// TODO:(niklasad1): Return `Result<(), Error>` here instead
 	fn populate_from_parent(&self, header: &mut Header, parent: &Header) {
 		// TODO(https://github.com/paritytech/parity-ethereum/issues/10410): this is a horrible hack,
 		// it is due to the fact that enact and miner both use OpenBlock::new() which will both call
 		// this function. more refactoring is definitely needed.
-		match header.extra_data().len() >= VANITY_LENGTH + SIGNATURE_LENGTH {
-			true => {
-				// we are importing blocks, do nothing.
-			}
-			false => {
-				trace!(target: "engine", "populate_from_parent in sealing");
-
-				// It's unclear how to prevent creating new blocks unless we are authorized, the best way (and geth does this too)
-				// it's just to ignore setting an correct difficulty here, we will check authorization in next step in generate_seal anyway.
-				if let Some(signer) = self.signer.read().as_ref() {
-					match self.state(&parent) {
-						Err(e) => {
-							trace!(target: "engine", "populate_from_parent: Unable to find parent state: {}, ignored.", e);
-						}
-						Ok(state) => {
-							if state.is_authorized(&signer.address()) {
-								if state.is_inturn(header.number(), &signer.address()) {
-									header.set_difficulty(DIFF_INTURN);
-								} else {
-									header.set_difficulty(DIFF_NOTURN);
-								}
-							}
-						}
+		if header.extra_data().len() < VANITY_LENGTH + SIGNATURE_LENGTH {
```

Comment:
only style cleanup to reduce the indentation level but it will fetch the state before trying to read `the signer`.

Also might be `out-of-scope` for this PR

Reply:
There doesn't seem to be any issue with this that I can identify off the top of my head.

Reply:
Actually.. this adds a bit of extra overhead that we don't need.  Can we move state query after the signer is read?

Reply:
Yeah, let's revert it